### PR TITLE
Add mention that subject could get claims from server side sessions.

### PR DESCRIPTION
**Pull request by RolandGuijt**

### CODE REVIEW

**Review by josephdecock — CHANGES_REQUESTED** [review]
I think we should incorporate this into the previous paragraph to make which claims you'll get more obvious. If server side sessions are on, we always use that. If they're off, then we give you different claims depending on the context.

**Comment by josephdecock** [comment]
And maybe we can explain the the claims identity at the time of grant creation is stored in the grant store. So if you are in the profile service in the context of using a grant (exchanging a code for a token, etc), we can give you those claims, but if you are in the user info endpoint, we don't have anything except what is in the access token sent to the user info endpoint.

**Comment by RolandGuijt** [comment]
@josephdecock I've used bullets now to create an if..then structure which should be more clear.

File: IdentityServer/v7/docs/content/reference/services/profile_service.md [hunk 1]
```diff
@@ -45,7 +45,12 @@ Models the request for user claims and is the vehicle to return those claims. It
 
 * ***Subject***
     
-    The *ClaimsPrincipal* modeling the user associated with this request for profile data. When the profile service is invoked for tokens, the *Subject* property will contain the principal that was issued during user sign-in. When the profile service is called for requests to the [userinfo endpoint]({{< ref "/reference/endpoints/userinfo" >}}), the *Subject* property will contain a claims principal populated with the claims in the access token used to authorize the userinfo call.
+    The *ClaimsPrincipal* modeling the user associated with this request for profile data. When the profile service is invoked for tokens, the *Subject* property will contain the user's principal. Which claims are contained in the principal depends on the following:
+
+    - When the [server side sessions feature]({{< ref "ui/server_side_sessions/" >}}) is enabled _Subject_ will always contain the claims stored in the server side session.
+    - When that is not the case, it depends on the caller context:
+        - If the _ProfileService_ is called in the context of a grant (e.g. exchanging a code for a token) the claims stored in the grant store will be used.
```
Comment (josephdecock) [inline IdentityServer/v7/docs/content/reference/services/profile_service.md:52]: This is good, but I think it raises another question - which claims are stored in the grant store?

Maybe something like this:

```suggestion
        - If the _ProfileService_ is called in the context of a grant (e.g. exchanging a code for a token), the claims associated with that grant in the grant store will be used. When grants are stored, by default a snapshot of the logged in user's claims are captured with the grant.
```

**Comment by RolandGuijt** [comment]
Agreed.

**Pull request merged**